angular2-jwt.ts

```diff
@@ -15,46 +15,54 @@ export interface IAuthConfig {
   noTokenScheme?:boolean;
 }
 
+// This interface should be used in the ctor of AuthHttp.
+export interface IAuthConfigOptional {
+    headerName?: string;
+    headerPrefix?: string;
+    tokenName?: string;
+    tokenGetter?: any;
+    noJwtError?: boolean;
+    globalHeaders?: Array<Object>;
+    noTokenScheme?: boolean;
+}
+
+class AuthConfigConsts {
+    public static readonly DEFAULT_TOKEN_NAME = 'id_token';
+    public static readonly DEFAULT_HEADER_NAME = 'Authorization';
+    public static readonly HEADER_PREFIX_BEARER = 'Bearer ';
+}
+
+const AuthConfigDefaults: IAuthConfig = {
+    headerName: AuthConfigConsts.DEFAULT_HEADER_NAME,
+    headerPrefix: null,
+    tokenName: AuthConfigConsts.DEFAULT_TOKEN_NAME,
+    tokenGetter: (() => localStorage.getItem(AuthConfigConsts.DEFAULT_TOKEN_NAME)),
+    noJwtError: false,
+    globalHeaders: [],
+    noTokenScheme: false
+};
+
 /**
  * Sets up the authentication configuration.
  */
 
 export class AuthConfig {
-  
-  headerName: string;
-  headerPrefix: string;
-  tokenName: string;
-  tokenGetter: any;
-  noJwtError: boolean;
-  noTokenScheme: boolean;
-  globalHeaders: Array<Object>;
 
-  constructor(config:any={}) {
-    this.headerName = config.headerName || 'Authorization';
-    if (config.headerPrefix) {
-      this.headerPrefix = config.headerPrefix + ' ';
-    } else if (config.noTokenScheme) {
-      this.headerPrefix = '';
+  private _config: IAuthConfig;
+
+  constructor(config: IAuthConfigOptional = AuthConfigDefaults) {
```

Comment:
Thanks for catching that. I've updated the PR.
I wonder, though, why it is not possible in typescript to express it the way I've done it.

For me it was just an optional object, which has itself optional fields with default values.
When I specify some of these fields, but not all, I would expect that the other fields use their default values, **but this is not the case**, as you @escardin have correctly pointed out.
I find this behavior really counterintuitive, but anyway... :confused: